src/Idris/Doc/Keywords.idr

```diff
@@ -505,31 +505,71 @@ withabstraction = vcat $
     ```
     """]
 
-letbinding : Doc IdrisDocAnn
-letbinding = vcat $
+-- to keep docs consistent between `:doc let` and `:doc (:=)`
+letBinding : String
+letBinding =
+    """
+    * Let bindings are defined using the `:=` syntax and can be used to bind the
+    result of intermediate computations. A type annotation can be added but is
+    not required. However, this results in different behaviours, for example:
+    ```idris
+    power4 : Nat -> Nat
+    power4 n = let square := n * n
+               in square * square
+
+    power4T : Nat -> Nat
+    power4T n = let square : Nat := n * n
+                in square * square
+    ```
+    In the first example, the type annotation is omitted and so `:=` gets
+    elaborated to the lambda expression `(\square => square * square) (n * n)`.
+
+    The second example does have a type annotation and as a result, `square`
+    gets computed rather than lambda-abstracted, meaning the result will be
+    reused on reference, instead of potentially being recomputed every time.
```

Comment:
I don't think that's true. The code for parsing let bindings is as follows:

```
    letBinder : Rule LetBinder
    letBinder = do s <- bounds (MkPair <$> multiplicity fname <*> expr plhs fname indents)
                   (rig, pat) <- pure s.val
                   ty <- option (PImplicit (virtualiseFC $ boundToFC fname s))
                                (decoratedSymbol fname ":" *> typeExpr (pnoeq pdef) fname indents)
                   (decoratedSymbol fname "=" <|> decoratedSymbol fname ":=")
                   val <- typeExpr pnowith fname indents
                   alts <- block (patAlt fname)
                   pure (MkLetBinder rig pat ty val alts)
```

As you can see if there is no type annotation, we replace it with a hole.

And so computationally the two expressions are the same.